packages/playwright-core/src/server/registry/browserFetcher.ts

```diff
@@ -19,15 +19,14 @@ import fs from 'fs';
 import os from 'os';
 import path from 'path';
 import childProcess from 'child_process';
-import { getUserAgent } from '../../utils/userAgent';
 import { existsAsync } from '../../utils/fileUtils';
 import { debugLogger } from '../../common/debugLogger';
-import { extract } from '../../zipBundle';
 import { ManualPromise } from '../../utils/manualPromise';
 import { colors } from '../../utilsBundle';
+import { browserDirectoryToMarkerFilePath } from '.';
 
 export async function downloadBrowserWithProgressBar(title: string, browserDirectory: string, executablePath: string | undefined, downloadURLs: string[], downloadFileName: string, downloadConnectionTimeout: number): Promise<boolean> {
-  if (await existsAsync(browserDirectory)) {
```

Comment:
this looked like a bug before, we didn't check for the marker file for some reason.

Reply:
Fair point. However, if the marker doesn't exist, you should remove the `browserDirectory` before trying to download something there.